brainframe_qt/ui/brainframe_app.py

```diff
@@ -1,98 +1,92 @@
 import logging
-import sys
 import typing
+import sys
 from traceback import TracebackException
 from typing import List, Optional
 
-from PyQt5.QtCore import QLocale, QMetaObject, QThread, QTranslator, Q_ARG, Qt, \
-    pyqtSlot
+from PyQt5.QtCore import QMetaObject, QThread, Q_ARG, Qt, pyqtSlot, QDeadlineTimer
 from PyQt5.QtGui import QIcon
 from PyQt5.QtWidgets import QApplication, QWidget
 
-from brainframe.api import bf_codecs, bf_errors
+from brainframe.api import bf_errors
 from gstly import gobject_init
 
 import brainframe_qt
 from brainframe_qt.api_utils import api
-from brainframe_qt.extensions.loader import ExtensionLoader
+from brainframe_qt.api_utils.connection_manager import ConnectionManager
 from brainframe_qt.ui import EULADialog, MainWindow, SplashScreen
-# noinspection PyUnresolvedReferences
 from brainframe_qt.ui.resources import QTAsyncWorker, qt_resources
 from brainframe_qt.ui.resources.config import ServerSettings
-from brainframe_qt.ui.resources.paths import text_paths
+from brainframe_qt.ui.resources.i18n.translator import BrainFrameTranslator
 from brainframe_qt.ui.resources.ui_elements.widgets.dialogs import BrainFrameMessage
-from brainframe_qt.util.secret import decrypt
 
 
 class BrainFrameApplication(QApplication):
     def __init__(self, argv: Optional[List] = ()):
         super().__init__(argv or [])
         self.server_config = ServerSettings()
 
-        self.splash_screen = typing.cast(SplashScreen, None)
+        self.connection_manager = ConnectionManager(parent=self)
+        self.splash_screen: SplashScreen = SplashScreen()
+        self.main_window: Optional[MainWindow] = None
 
-        # noinspection PyUnresolvedReferences
-        self.aboutToQuit.connect(self._shutdown)
-
-        self.setWindowIcon(QIcon(":/icons/window_icon"))
-        self.setOrganizationDomain('aotu.ai')
+        self.translator = self._init_translator()
 
-        self._init_translator()
+        self._init_style()
+        self._init_signals()
 
+        self._init_config()
         gobject_init.start(start_main_loop=False)
 
-        self._init_server_settings()
-
-    def _init_translator(self):
```

Comment:
So what changed that allows you to get rid of this logic?

Reply:
I see, you moved it all into it's own class. Was there any change to the logic at all?

Reply:
No functional changes should have occured